File: myddd-extensions/myddd-security/myddd-security-infra/src/test/java/org/myddd/extensions/security/domain/UserRepositoryTest.java
package org.myddd.extensions.security.domain;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.myddd.extensions.security.AbstractTest;

import javax.inject.Inject;
import javax.transaction.Transactional;
import java.util.List;

class UserRepositoryTest extends AbstractTest {

    @Inject
    private UserRepository userRepository;

    @Test
    void testUserRepositoryNotNull(){
        Assertions.assertNotNull(userRepository);
    }

    @Test
    @Transactional
    void testQueryUserById(){
        User notExistUser = userRepository.queryLocalUserByUserId(randomId());
        Assertions.assertNull(notExistUser);

        User created = userRepository.save(randomUser());
        Assertions.assertNotNull(created);

        User query = userRepository.queryLocalUserByUserId(created.getUserId());
        Assertions.assertNotNull(query);
    }

    @Test
    @Transactional
    void testQueryUserByEmail(){
        Assertions.assertNull(userRepository.queryUserByEmail(randomId(),UserType.LOCAL));

        User created = userRepository.save(randomUser());
        Assertions.assertNotNull(created);
        Assertions.assertNotNull(userRepository.queryUserByEmail(created.getEmail(),UserType.LOCAL));
    }

    @Test
    @Transactional
    void testQueryUserByPhone(){
        Assertions.assertNull(userRepository.queryUserByPhone(randomId(),UserType.LOCAL));

        var user = randomUser();
        user.setPhone("18620501006");
        var createdUser = user.createLocalUser();

        Assertions.assertNotNull(userRepository.queryUserByPhone(createdUser.getPhone(),UserType.LOCAL));
    }

    @Test
    @Transactional
    void testBatchInsertUsers(){
        var list = List.of(randomUser(),randomUser());
        userRepository.batchSaveEntities(list);

        var userIdList = userRepository.queryExistsUserIdByType(UserType.LOCAL);
        Assertions.assertFalse(userIdList.isEmpty());
    }
}

File: examples/kotlin-ddd/infra/src/test/kotlin/org/myddd/kotlin/infra/UserRepositoryTest.kt
package org.myddd.kotlin.infra

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.myddd.domain.InstanceFactory
import org.myddd.kotlin.domain.UserRepository
import org.myddd.kotlin.AbstractTest
import javax.transaction.Transactional

class UserRepositoryTest: AbstractTest() {

    private val userRepository by lazy { InstanceFactory.getInstance(UserRepository::class.java) }

    @Test
    @Transactional
    fun createUser(){
        val randomUser = randomUser()
        val createUser = userRepository.createUser(randomUser)
        Assertions.assertNotNull(createUser)
    }

    @Test
    @Transactional
    fun searchUser(){
        Assertions.assertTrue(userRepository.searchUserByName(randomString()).isEmpty())

        val createUser = userRepository.createUser(randomUser())
        Assertions.assertNotNull(createUser)
        Assertions.assertTrue(userRepository.searchUserByName(createUser!!.name).isNotEmpty())
    }
}